pkg/agent/agent.go

```diff
@@ -347,6 +357,20 @@ func (k *Klocksmith) getPodsForDeletion() ([]v1.Pod, error) {
 	return pods, nil
 }
 
+// waitForPodDeletion waits for a pod to be deleted
+func (k *Klocksmith) waitForPodDeletion(pod v1.Pod) error {
+	return wait.PollInfinite(time.Second, func() (bool, error) {
+		switch _, err := k.kc.CoreV1().Pods(pod.Namespace).Get(pod.Name, v1meta.GetOptions{}); {
+		case err == nil:
+			return false, nil
+		case errors.IsNotFound(err):
+			return true, nil
+		default:
```

Comment:
Handling this more naively could improve logs and retry. Compared with logging at the top level where the wait return error is just a loose approximation of what happened.

```
_, err := k.kc.CoreV1().Pods(pod.Namespace).Get(pod.Name, v1meta.GetOptions{})
if errors.IsNotFound(err) {
    glog.Infof("deleted pod %s", pod.Name)   // success log
	return true, nil
}
if err != nil {
  glog.Errorf("failed to get pod %v", err)    // some API call error, likely transient
}
return false, nil
```

```
Waiting for pod POD to terminate
failed to get pod: transient error
failed to get pod: transient error
deleted pod POD
```

or 

```
Waiting for pod POD to terminate
failed to get pod: transient error
failed to get pod: transient error
Skipping wait on pod POD
```